File: Excepciones/InformationException.vb
Public Class InformationException
    Inherits Exception

    Public Property ListaMensajes As List(Of String)

    Public Sub New(ByVal mensaje As String)
        MyBase.New(mensaje)
    End Sub

    Public Sub New(ByVal Mensajes As List(Of String))
        ListaMensajes = Mensajes
    End Sub
End Class

File: Excepciones/InformationException.cs
using System;
using System.Collections.Generic;

namespace Excepciones
{
    public class InformationException : Exception
    {
        public List<string> ListaMensajes { get; set; }

        public InformationException(string mensaje) : base(mensaje)
        {
        }

        public InformationException(List<string> Mensajes)
        {
            ListaMensajes = Mensajes;
        }
    }
}